test/lit/passes/nontrapping-fptoint-lowering.wast

```diff
@@ -0,0 +1,208 @@
+ ;; CHECK-NEXT:     (f32.ge
+ ;; CHECK-NEXT:      (local.get $6)
+ ;; CHECK-NEXT:      (f32.const 0)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (then
+ ;; CHECK-NEXT:     (i32.trunc_f32_u
+ ;; CHECK-NEXT:      (local.get $6)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (else
+ ;; CHECK-NEXT:     (i32.const 0)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (if (result i64)
+ ;; CHECK-NEXT:    (i32.and
+ ;; CHECK-NEXT:     (f32.lt
+ ;; CHECK-NEXT:      (local.tee $7
+ ;; CHECK-NEXT:       (local.get $0)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (f32.const 18446744073709551615)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:     (f32.ge
+ ;; CHECK-NEXT:      (local.get $7)
+ ;; CHECK-NEXT:      (f32.const 0)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (then
+ ;; CHECK-NEXT:     (i64.trunc_f32_u
+ ;; CHECK-NEXT:      (local.get $7)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (else
+ ;; CHECK-NEXT:     (i64.const 0)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (if (result i32)
+ ;; CHECK-NEXT:    (i32.and
+ ;; CHECK-NEXT:     (f64.lt
+ ;; CHECK-NEXT:      (local.tee $8
+ ;; CHECK-NEXT:       (local.get $1)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (f64.const 4294967295)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:     (f64.ge
+ ;; CHECK-NEXT:      (local.get $8)
+ ;; CHECK-NEXT:      (f64.const 0)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (then
+ ;; CHECK-NEXT:     (i32.trunc_f64_u
+ ;; CHECK-NEXT:      (local.get $8)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (else
+ ;; CHECK-NEXT:     (i32.const 0)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (drop
+ ;; CHECK-NEXT:   (if (result i64)
+ ;; CHECK-NEXT:    (i32.and
+ ;; CHECK-NEXT:     (f64.lt
+ ;; CHECK-NEXT:      (local.tee $9
+ ;; CHECK-NEXT:       (local.get $1)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (f64.const 18446744073709551615)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:     (f64.ge
+ ;; CHECK-NEXT:      (local.get $9)
+ ;; CHECK-NEXT:      (f64.const 0)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (then
+ ;; CHECK-NEXT:     (i64.trunc_f64_u
+ ;; CHECK-NEXT:      (local.get $9)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (else
+ ;; CHECK-NEXT:     (i64.const 0)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $truncsat  (type $0)
+  (local $0 f32)
+  (local $1 f64)
+  (drop (i32.trunc_sat_f32_s (local.get $0)))
+  (drop (i64.trunc_sat_f32_s (local.get $0)))
+  (drop (i32.trunc_sat_f64_s (local.get $1)))
+  (drop (i64.trunc_sat_f64_s (local.get $1)))
+  (drop (i32.trunc_sat_f32_u (local.get $0)))
+  (drop (i64.trunc_sat_f32_u (local.get $0)))
+  (drop (i32.trunc_sat_f64_u (local.get $1)))
+  (drop (i64.trunc_sat_f64_u (local.get $1)))
```

Comment:
Maybe put each of these in its own function? We try to do that so the the before and after are close together.

Reply:
Yeah, could do that. This was pasted and edited from the upstream core wast test, but I could probably massage it some more.

Reply:
Oh, if it's from upstream, maybe it's best to keep it.